File: app/src/main/java/me/shreyasr/chatse/event/ChatEventGenerator.java
package me.shreyasr.chatse.event;

import org.codehaus.jackson.JsonNode;
import org.codehaus.jackson.map.ObjectMapper;

import java.io.IOException;

import me.shreyasr.chatse.util.Logger;

public class ChatEventGenerator {

    private ObjectMapper mapper = new ObjectMapper();

    public ChatEvent createEvent(JsonNode json) {
        try {
            return mapper.readValue(json, ChatEvent.class);
        } catch (IOException e) {
            Logger.exception(this.getClass(), "Failed to map json", e);
        }
        return null;
    }
}

File: app/src/main/java/me/shreyasr/chatse/event/ChatEventGenerator.kt
package me.shreyasr.chatse.event

import org.codehaus.jackson.JsonNode
import org.codehaus.jackson.map.ObjectMapper
import timber.log.Timber
import java.io.IOException

class ChatEventGenerator {

    private val mapper = ObjectMapper()

    fun createEvent(json: JsonNode): ChatEvent? {
        var c: ChatEvent? = null
        try {
            c = mapper.readValue<ChatEvent>(json, ChatEvent::class.java)
        } catch (e: IOException) {
            Timber.e("Failed to map json", e)
        }

        return c
    }
}
